src/page-modules/assistant/trip-pattern/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { formatLocaleTime, secondsToDuration } from '@atb/utils/date';
+import { type TripPattern } from '@atb/page-modules/assistant/server/journey-planner/validators';
+import style from './trip-pattern.module.css';
+import { ComponentText, useTranslation } from '@atb/translations';
+import { Typo } from '@atb/components/typography';
+import { motion } from 'framer-motion';
+import { Button } from '@atb/components/button';
+import { MonoIcon } from '@atb/components/icon';
+import { TransportIconWithLabel } from '@atb/components/transport-mode/transport-icon';
+
+type TripPatternProps = {
+  tripPattern: TripPattern;
+  index: number;
+};
+export function TripPattern({ tripPattern, index }: TripPatternProps) {
+  const { t, language } = useTranslation();
+
+  const duration = secondsToDuration(tripPattern.duration, language);
+  const fromPlace = tripPattern.legs[0].fromPlace.name ?? '';
+
+  return (
+    <motion.div
+      className={style.tripPattern}
+      initial={{ opacity: 0, x: -10 }}
+      animate={{ opacity: 1, x: 0 }}
+      exit={{ opacity: 0, x: -10 }}
+      transition={{
+        delay: index * 0.1, // staggerChildren on parent only works first render
+      }}
+    >
+      <header className={style.header}>
+        <Typo.span textType="body__secondary--bold">
+          {t(ComponentText.TripPattern.busFrom)} {fromPlace}
+        </Typo.span>
+        <Typo.span
+          textType="body__secondary"
+          className={style.header__duration}
+        >
+          {duration}
+        </Typo.span>
+      </header>
+
+      <div className={style.legs}>
+        {tripPattern.legs.map((leg, i) => (
+          <div
+            key={`leg-${leg.expectedStartTime}-${i}`}
+            className={style.legs__leg}
+          >
+            <div className={style.legs__leg__icon}>
+              {leg.mode ? (
+                <TransportIconWithLabel
+                  mode={{ mode: leg.mode }}
+                  label={leg.line?.publicCode}
+                />
+              ) : (
+                <div className={style.legs__leg__walkIcon}>
+                  <MonoIcon icon="transportation/Walk" />
+                </div>
+              )}
+            </div>
+
+            <Typo.span textType="body__tertiary">
+              {formatLocaleTime(tripPattern.expectedEndTime, language)}
+            </Typo.span>
+          </div>
+        ))}
```

Comment:
Maybe this should be wrapped in a container which has `overflow: hidden`? Think this will solve most cases on smaller screens 🤔

Reply:
Can do that, taking that into account when optimizing for mobile. According to the design we shall only show the 3(?) first legs as well, and truncate the rest into a count only.

Reply:
Is this always true no matter size? Is it simplification or just practical? Should talk to Julia I guess

Reply:
Should talk to Julia, yes.